gobblin-compaction/src/main/java/gobblin/compaction/hivebasedconstructs/HiveMetadataForCompactionExtractor.java

```diff
@@ -0,0 +1,91 @@
+package gobblin.compaction.hivebasedconstructs;
+
+import java.io.IOException;
+import org.apache.avro.Schema;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hive.metastore.IMetaStoreClient;
+import org.apache.hadoop.hive.metastore.api.Table;
+import org.apache.hadoop.hive.ql.metadata.HiveException;
+import org.apache.thrift.TException;
+import com.google.common.base.Optional;
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.configuration.WorkUnitState;
+import gobblin.data.management.conversion.hive.avro.AvroSchemaManager;
+import gobblin.data.management.conversion.hive.source.HiveWorkUnit;
+import gobblin.data.management.conversion.hive.watermarker.PartitionLevelWatermarker;
+import gobblin.data.management.copy.hive.HiveDataset;
+import gobblin.data.management.copy.hive.HiveDatasetFinder;
+import gobblin.hive.HiveMetastoreClientPool;
+import gobblin.source.extractor.Extractor;
+import gobblin.util.AutoReturnableObject;
+import lombok.extern.slf4j.Slf4j;
+
+
+/**
+ * {@link Extractor} that extracts primary key field name, delta field name, and location from hive metastore and
+ * creates an {@link MRCompactionEntity}
+ */
+@Slf4j
+public class HiveMetadataForCompactionExtractor implements Extractor<Schema, MRCompactionEntity> {
+
+  public static final String COMPACTION_PRIMARY_KEY = "hive.metastore.primaryKey";
+  public static final String COMPACTION_DELTA = "hive.metastore.delta";
+
+  private MRCompactionEntity compactionEntity;
+  private boolean extracted = false;
+
+  public HiveMetadataForCompactionExtractor(WorkUnitState state, FileSystem fs) throws IOException, TException, HiveException {
+    if (Boolean.valueOf(state.getPropAsBoolean(PartitionLevelWatermarker.IS_WATERMARK_WORKUNIT_KEY))) {
+      log.info("Ignoring Watermark workunit for {}", state.getProp(ConfigurationKeys.DATASET_URN_KEY));
+      return;
+    }
+
+    HiveWorkUnit hiveWorkUnit = new HiveWorkUnit(state.getWorkunit());
```

Comment:
I see copy/paste code from HiveConvertExtractor. You should refactor and reuse code (extend HiveConvertExtractor). Ask yourself, what makes a Hive extractor different than any other extractor.